### ISSUE TITLE
Add format of ISO week-numbering year

### ISSUE THREAD

**Issue by wenqi73** [angular/angular#38739]
# 🚀 feature request

### Relevant Package

 This feature request is for @angular/common

### Description

```ts
formatDate(new Date(2018, 11, 31), 'yyyy-ww', 'en-US')
// 2018-01
```
Week is right but year is wrong.
### Describe the solution you'd like

Add an ISO week-numbering year like `rrrr`.
```ts
formatDate(new Date(2018, 11, 31), 'rrrr-ww', 'en-US')
// 2019-01
```
Here is the reference of [date-fns](https://date-fns.org/v2.16.1/docs/format).

**Comment by petebacondarwin** [comment]
Normally I am cautious about adding new functionality but I think that this case could be an exception. Since we are already doing the calculations for the week numbering, it is probably not too much more to compute the "week year".

This might be a good first time PR for someone... 🤔

**Comment by prashant-tholia** [comment]
Hi, I'd like to work on this issue.

**Comment by petebacondarwin** [comment]
@prashant-tholia - great! Please make sure you have read https://github.com/angular/angular/blob/master/CONTRIBUTING.md.
The PR will need to include unit tests and documentation before it can be merged.

Let me know if you need help getting started.

**Comment by prashant-tholia** [comment]
Thanks @petebacondarwin! I'll make sure to fulfill all mentioned guidelines and requirements.

**Comment by irzakamran** [comment]
Hello, can I also be work on this issue with @prashant-tholia since this is a good first issue and I am an open source beginner.

**Comment by rubenprins** [comment]
@prashanttholia, Not to be pedantic or anything, but I think the 'r' placeholder is a mistake here. CLDR/Unicode defines this in http://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table as

> Related Gregorian year (numeric). For non-Gregorian calendars, this corresponds to the extended Gregorian year in which the calendar’s year begins. Related Gregorian years are often displayed, for example, when formatting dates in the Japanese calendar — e.g. “2012(平成24)年1月15日” — or in the Chinese calendar — e.g. “2012壬辰年腊月初四”. The related Gregorian year is usually displayed using the "latn" numbering system, regardless of what numbering systems may be used for other parts of the formatted date. If the calendar’s year is linked to the solar year (perhaps using leap months), then for that calendar the ‘r’ year will always be at a fixed offset from the ‘u’ year. For the Gregorian calendar, the ‘r’ year is the same as the ‘u’ year. For ‘r’, all field lengths specify a minimum number of digits; there is no special interpretation for “rr”.

The ISO week year is actually 'Y', 'YY', 'YYY', 'YYYY', etc.:

> Year in “Week of Year” based calendars in which the year transition occurs on a week boundary; may differ from calendar year ‘y’ near a year transition. This numeric year designation is used in conjunction with pattern character ‘w’ in the ISO year-week calendar as defined by ISO 8601, but can be used in non-Gregorian based calendar systems where week date processing is desired. The field length is interpreted in the same was as for ‘y’; that is, “yy” specifies use of the two low-order year digits, while any other field length specifies a minimum number of digits to display.

**Comment by prashanttholia** [comment]
@rubenprins, I apologize for the late response. I think that formats should be consistent with or as close as possible to the standard. I agree that 'Y' would be a more appropriate choice to represent week numbering year. 

@petebacondarwin, please let us know what you think and whether it needs to be updated?

**Comment by petebacondarwin** [comment]
@prashanttholia  - sorry for not replying sooner. We discussed this and felt it would be better to go with the `Y` formatting character rather than `r`. I know that you are keen to help out in this but since we are really close to final release of 11.0.0, @AndrewKushnir went ahead and created a PR to fix this up in https://github.com/angular/angular/pull/39495. Thank you for the original PR.

(We need to work out what to do about ISO alignment for weeks etc too.)

(Also thanks to @cexbrayat for pinging me about this outstanding question!)

**Comment by prashanttholia** [comment]
@petebacondarwin, no worries. I am really glad that it got fixed. 

I have mentioned my thoughts on the ISO alignment issue in the comment [here](https://github.com/angular/angular/issues/25380#issuecomment-719924159). Please have a look.

**Comment by petebacondarwin** [comment]
Thanks @prashanttholia

**Comment by abadi003** [comment]
is this issue still open?

**Comment by AndrewKushnir** [comment]
@abadi003 this issue is resolved by #38828, so I'm closing this ticket. Thank you.
